java/runtime-common/src/main/java/org/ray/core/RayRuntime.java

```diff
@@ -357,69 +349,53 @@ public RayObjects call(UniqueID taskId, Class<?> funcCls, RayFunc lambda, int re
         localSchedulerProxy.notifyUnblocked();
       }
     }
-
   }
 
   private <T> T doGet(UniqueID objectId, boolean isMetadata) throws TaskExecutionException {
+    ImmutableList<UniqueID> objectIds = ImmutableList.of(objectId);
+    List<T> results = doGet(objectIds, isMetadata);
 
-    boolean wasBlocked = false;
-    UniqueID taskId = getCurrentTaskId();
-    try {
-      // Do an initial fetch.
-      objectStoreProxy.fetch(objectId);
-
-      // Get the object. We initially try to get the object immediately.
-      Pair<T, GetStatus> ret = objectStoreProxy
-          .get(objectId, params.default_first_check_timeout_ms, isMetadata);
-
-      wasBlocked = (ret.getRight() != GetStatus.SUCCESS);
-
-      // Try reconstructing the object. Try to get it until at least PlasmaLink.GET_TIMEOUT_MS
-      // milliseconds passes, then repeat.
-      while (ret.getRight() != GetStatus.SUCCESS) {
-        RayLog.core.warn(
-            "Task " + taskId + " Object " + objectId.toString() + " get failed, reconstruct ...");
-        // Do another fetch
-        localSchedulerProxy.reconstructObject(objectId, false);
-
-        if (!params.use_raylet) {
-          objectStoreProxy.fetch(objectId);
-        }
-
-        //Check the result every 5s, but it will return once available.
-        ret = objectStoreProxy.get(objectId, params.default_get_check_interval_ms,
-            isMetadata);
-      }
-      RayLog.core.warn(
-          "Task " + taskId + " Object " + objectId.toString() + " get" + ", the result " + ret
-          .getLeft());
-      return ret.getLeft();
-    } catch (TaskExecutionException e) {
-      RayLog.core
-          .error("Task " + taskId + " Object " + objectId.toString() + " get with Exception", e);
-      throw e;
-    } finally {
-      // If the object was not able to get locally, let the local scheduler
-      // know that we're now unblocked.
-      if (wasBlocked) {
-        localSchedulerProxy.notifyUnblocked();
-      }
-    }
-
+    assert results.size() == 1;
+    return results.get(0);
```

Comment:
minor, this can be simplified to one line `return doGet(ImmutableList(objectId, isMetadata).get(0)`.
but if you prefer the current way, it's fine as well.